### PR TITLE
Limit available payment methods by contract kind

### PR DESCRIPTION
**Pull request by Raibo**
https://github.com/happy-travel/agent-app-project/issues/113

### CODE REVIEW

**Review by AlexeyShok — APPROVED** [review]

File: Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs [hunk 1]
```diff
@@ -49,6 +53,10 @@ public class BookingEvaluationService : IBookingEvaluationService
 
             var originalSupplierPrice = connectorEvaluationResult.Value?.RoomContractSet.Rate.FinalPrice ?? default;
             
+            var (_, isContractFailure, contractKind, contractError) = await _counterpartyService.GetContractKind(agent.CounterpartyId);
+            if (isContractFailure)
+                return ProblemDetailsBuilder.Fail<RoomContractSetAvailability?>(contractError);
```
Comment (mingazhev) [inline Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs:58]: Don't think that this method can really fail. Maybe not check the result or throw an exception? Existing AgentContext guarantees that we have counterparty

File: Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingPaymentMethodsHelper.cs [hunk 2]
```diff
@@ -2,17 +2,19 @@
...
 {
     public static class BookingPaymentMethodsHelper
     {
         public static List<PaymentMethods> GetAvailablePaymentMethods(in EdoContracts.Accommodations.RoomContractSetAvailability availability,
-            AccommodationBookingSettings settings, DateTime date)
+            AccommodationBookingSettings settings, CounterpartyContractKind contractKind, DateTime date)
             => AllAvailablePaymentMethods
                 .Intersect(GetAprPaymentMethods(availability, settings))
                 .Intersect(GetPassedDeadlinePaymentMethods(availability, settings, date))
+                .Intersect(GetConytactKindPaymentMethods(contractKind))
```
Comment (mingazhev) [inline Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingPaymentMethodsHelper.cs:17]: Typo in method name

File: Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs (hunk 1 above)
Comment (Raibo) [reply Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs:58]: Is it also guaranteed that `ContractKind` is not null there?

File: Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingPaymentMethodsHelper.cs (hunk 2 above)
Comment (Raibo) [reply Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingPaymentMethodsHelper.cs]: Fixed

File: Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs:58]: No, it is not guaranteed. Ok, let's leave this

**Review by mingazhev — APPROVED** [review]

**Pull request merged**